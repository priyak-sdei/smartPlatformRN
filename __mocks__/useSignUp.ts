// __mocks__/useSignUp.ts

let mockState = {
    name: '',
    setName: jest.fn(),
    email: '',
    setEmail: jest.fn(),
    password: '',
    setPassword: jest.fn(),
    confirmPassword: '',
    setConfirmPassword: jest.fn(),
    handleSignUp: jest.fn(),
    isLoading: false,
    error: null,
    isValid: () => false,
};

export const __setMockSignUpState = (state: Partial<typeof mockState>) => {
    mockState = { ...mockState, ...state };
};

export const useSignUp = () => ({
    ...mockState,
    isValid: typeof mockState.isValid === 'function' ? mockState.isValid : () => mockState.isValid,
}); 