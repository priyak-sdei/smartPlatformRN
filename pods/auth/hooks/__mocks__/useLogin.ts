// __mocks__/useLogin.ts

const defaultState = {
    isValid: false,
    loading: false,
    error: null,
    values: { email: '', password: '' },
    setEmail: jest.fn(),
    setPassword: jest.fn(),
    onLogin: jest.fn(),
};

let mockState = {
    isValid: false,
    loading: false,
    error: null,
    email: '',
    password: '',
    setEmail: jest.fn(),
    setPassword: jest.fn(),
    login: jest.fn(),
};

export const __setMockLoginState = (state: Partial<typeof mockState>) => {
    mockState = { ...mockState, ...state };
};

export const resetMockLoginState = () => {
    mockState = {
        isValid: false,
        loading: false,
        error: null,
        email: '',
        password: '',
        setEmail: jest.fn(),
        setPassword: jest.fn(),
        login: jest.fn(),
    };
};

export const useLogin = () => ({
    ...mockState,
    isValid: () => mockState.isValid,
}); 