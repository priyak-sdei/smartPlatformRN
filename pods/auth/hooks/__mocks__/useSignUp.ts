// __mocks__/useSignUp.ts

let mockState = {
    isValid: false,
    loading: false,
    error: null,
    values: { email: '', password: '', confirmPassword: '' },
    setEmail: jest.fn(),
    setPassword: jest.fn(),
    setConfirmPassword: jest.fn(),
    onSignUp: jest.fn(),
};

export const __setMockSignUpState = (state: Partial<typeof mockState>) => {
    mockState = { ...mockState, ...state };
};

export const useSignUp = () => ({
    ...mockState,
    isValid: () => mockState.isValid,
}); 