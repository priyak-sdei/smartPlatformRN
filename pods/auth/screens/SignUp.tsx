import type { AuthNavigationProp } from '../types';
import { Button, Layout, Text, TextInput } from '@shared/index';
import { Back } from '@shared/theme';
import {
  moderateScale,
  spacing,
  verticalScale,
  fonts,
  useTheme,
} from '@shared/theme';
import React from 'react';
import { useTranslation } from 'react-i18next';
import { StyleSheet } from 'react-native';
import { useSignUp } from '../hooks/useSignUp';

interface SignUpProps {
  navigation: AuthNavigationProp<'SignUp'>;
  useSignUpHook?: typeof useSignUp;
}

const SignUp = ({ navigation: _navigation, useSignUpHook = useSignUp }: SignUpProps) => {
  const {
    name,
    setName,
    email,
    setEmail,
    password,
    setPassword,
    confirmPassword,
    setConfirmPassword,
    handleSignUp,
    isValid,
  } = useSignUpHook();
  const { t } = useTranslation();
  const { theme } = useTheme();
  const styles = getStyles(theme);
  return (
    <Layout>
      <Layout.Header
        leftIcon={<Back />}
        onLeftPress={() => _navigation.goBack()}
        leftTestID="header-left"
      />
      <Layout.Body scrollable={true}>
        <Text variant="title" tx="login.signUp" style={styles.titleMargin} />
        <TextInput
          value={name}
          onChangeText={setName}
          autoCapitalize="words"
          labelTx="auth.name"
          placeholderTx={'auth.name'}
        />
        <TextInput
          value={email}
          onChangeText={setEmail}
          keyboardType="email-address"
          labelTx="login.email"
          placeholderTx={'login.email'}
        />
        <TextInput
          value={password}
          onChangeText={setPassword}
          secureTextEntry={true}
          labelTx="auth.createPassword"
          placeholderTx={'auth.createPassword'}
        />
        <TextInput
          value={confirmPassword}
          onChangeText={setConfirmPassword}
          secureTextEntry={true}
          labelTx="auth.confirmPassword"
          placeholderTx={'auth.confirmPassword'}
        />

        <Button
          tx={'login.signUp'}
          style={styles.button}
          onPress={handleSignUp}
          disabled={!isValid()}
        />

        <Text
          style={styles.bottomText}
          onPress={() => _navigation.navigate('Login')}
        >
          {t('login.alreadyAccount')}{' '}
          <Text style={styles.bottomTextBold}>{t('login.signIn')}</Text>
        </Text>
      </Layout.Body>
    </Layout>
  );
};

const getStyles = (theme: any) =>
  StyleSheet.create({
    button: {
      marginTop: verticalScale(spacing.m),
    },
    titleMargin: {
      marginBottom: moderateScale(spacing.s),
    },
    bottomText: {
      textAlign: 'center',
      marginTop: verticalScale(spacing.xxs),
      color: theme.colors.secondary,
      fontSize: moderateScale(spacing.s),
    },
    bottomTextBold: {
      fontFamily: fonts.medium,
    },
  });

export default SignUp;
