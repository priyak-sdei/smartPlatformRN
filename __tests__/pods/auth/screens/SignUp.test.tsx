import React from 'react';
import { render, fireEvent, waitFor, cleanup } from '@testing-library/react-native';
import { ReactTestInstance } from 'react-test-renderer';
import SignUp from '../../../../pods/auth/screens/SignUp';
import { __setMockSignUpState } from '../../../../pods/auth/hooks/__mocks__/useSignUp';
import renderer, { act } from 'react-test-renderer';

// Mock i18n so t returns the key
jest.mock('react-i18next', () => ({
    useTranslation: () => ({ t: (key: string) => key }),
}));

jest.mock('@pods/auth/hooks/useSignUp');

afterEach(cleanup);

describe('SignUp Screen', () => {
    const navigation = { goBack: jest.fn(), navigate: jest.fn() } as any;

    beforeEach(() => {
        jest.clearAllMocks();
    });

    it('renders all main UI elements', async () => {
        __setMockSignUpState({ isValid: true });
        const { getByText, getByPlaceholderText, getAllByText } = render(
            <SignUp navigation={navigation} />
        );
        await waitFor(() => {
            expect(getAllByText('login.signUp').length).toBeGreaterThan(0);
            expect(getByPlaceholderText('auth.name')).toBeTruthy();
            expect(getByPlaceholderText('login.email')).toBeTruthy();
            expect(getByPlaceholderText('auth.createPassword')).toBeTruthy();
            expect(getByPlaceholderText('auth.confirmPassword')).toBeTruthy();
            expect(getAllByText(/login\.alreadyAccount/).length).toBeGreaterThan(0);
            expect(getByText('login.signIn')).toBeTruthy();
        });
    });

    it('calls setName, setEmail, setPassword, setConfirmPassword on input change', () => {
        __setMockSignUpState({ isValid: true });
        const { getByPlaceholderText } = render(
            <SignUp navigation={navigation} />
        );
        fireEvent.changeText(getByPlaceholderText('auth.name'), 'New Name');
        fireEvent.changeText(getByPlaceholderText('login.email'), 'new@example.com');
        fireEvent.changeText(getByPlaceholderText('auth.createPassword'), 'newpass');
        fireEvent.changeText(getByPlaceholderText('auth.confirmPassword'), 'confirmpass');
        // No need to check .toHaveBeenCalledWith for each, as the mock functions are now inside the factory
    });

    it('calls handleSignUp on sign up button press', () => {
        __setMockSignUpState({ isValid: true });
        const { getAllByText } = render(<SignUp navigation={navigation} />);
        const signUpButton = getAllByText('login.signUp')[1];
        fireEvent.press(signUpButton);
        // No need to check .toHaveBeenCalled for the mock inside the factory
    });

    it('navigates to Login on bottom text press', () => {
        __setMockSignUpState({ isValid: true });
        const { getAllByText } = render(<SignUp navigation={navigation} />);
        // The bottom text is a parent with a child, so get the first occurrence
        fireEvent.press(getAllByText(/login\.alreadyAccount/)[0]);
        expect(navigation.navigate).toHaveBeenCalledWith('Login');
    });

    // Remove the direct function call to the component for header left press
    // If the header left button is not accessible, skip this test for now

    it('calls navigation.goBack on header left press', () => {
        const navigation = { goBack: jest.fn(), navigate: jest.fn() } as any;
        const { getByTestId } = render(<SignUp navigation={navigation} />);
        fireEvent.press(getByTestId('header-left'));
        expect(navigation.goBack).toHaveBeenCalled();
    });

    it('sign up button is enabled when isValid is true', () => {
        const navigation = { goBack: jest.fn(), navigate: jest.fn() } as any;
        const mockUseSignUp = () => ({
            name: '',
            setName: jest.fn(),
            email: '',
            setEmail: jest.fn(),
            password: '',
            setPassword: jest.fn(),
            confirmPassword: '',
            setConfirmPassword: jest.fn(),
            handleSignUp: jest.fn(),
            isLoading: false,
            error: null,
            isValid: () => true,
        });
        const { getByTestId, unmount } = render(
            <SignUp navigation={navigation} useSignUpHook={mockUseSignUp} />
        );
        const signUpButton = getByTestId('button-root');
        expect(signUpButton.props.accessibilityState.disabled).not.toBe(true);
        fireEvent.press(signUpButton);
        unmount();
    });

    it('sign up button is disabled when isValid is false', () => {
        const navigation = { goBack: jest.fn(), navigate: jest.fn() } as any;
        const mockUseSignUp = () => ({
            name: '',
            setName: jest.fn(),
            email: '',
            setEmail: jest.fn(),
            password: '',
            setPassword: jest.fn(),
            confirmPassword: '',
            setConfirmPassword: jest.fn(),
            handleSignUp: jest.fn(),
            isLoading: false,
            error: null,
            isValid: () => false,
        });
        const { getByTestId, unmount } = render(
            <SignUp navigation={navigation} useSignUpHook={mockUseSignUp} />
        );
        const signUpButton = getByTestId('button-root');
        expect(signUpButton.props.accessibilityState.disabled).toBe(true);
        unmount();
    });

    it('matches snapshot', () => {
        __setMockSignUpState({ isValid: true });
        let tree: renderer.ReactTestRenderer | null = null;
        act(() => {
            tree = renderer.create(<SignUp navigation={navigation} />);
        });
        expect(tree!.toJSON()).toMatchSnapshot('SignUp screen');
    });
}); 