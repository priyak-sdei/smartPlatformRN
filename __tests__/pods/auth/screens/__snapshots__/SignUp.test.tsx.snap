// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SignUp Screen matches snapshot: SignUp screen 1`] = `
<RNCSafeAreaView
  edges={
    {
      "bottom": "off",
      "left": "additive",
      "right": "additive",
      "top": "additive",
    }
  }
  style={
    [
      {
        "backgroundColor": "#FFFFFF",
        "flex": 1,
      },
      undefined,
    ]
  }
>
  <View
    style={
      {
        "flex": 1,
      }
    }
  >
    <View
      style={
        [
          {
            "backgroundColor": "#FFFFFF",
            "flexDirection": "row",
            "height": 75,
            "justifyContent": "space-between",
            "paddingHorizontal": 24,
          },
          undefined,
          undefined,
        ]
      }
    >
      <View
        style={
          {
            "alignItems": "center",
            "justifyContent": "center",
            "width": 60,
          }
        }
      >
        <View
          accessibilityState={
            {
              "busy": undefined,
              "checked": undefined,
              "disabled": undefined,
              "expanded": undefined,
              "selected": undefined,
            }
          }
          accessibilityValue={
            {
              "max": undefined,
              "min": undefined,
              "now": undefined,
              "text": undefined,
            }
          }
          accessible={true}
          collapsable={false}
          focusable={true}
          hitSlop={
            {
              "bottom": 10,
              "left": 10,
              "right": 10,
              "top": 10,
            }
          }
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            {
              "opacity": 1,
            }
          }
          testID="header-left"
        >
          <test-file-stub />
        </View>
      </View>
      <View
        style={
          {
            "alignItems": "center",
            "flex": 1,
            "justifyContent": "center",
            "minWidth": 0,
          }
        }
      />
      <View
        style={
          {
            "alignItems": "center",
            "justifyContent": "center",
            "width": 60,
          }
        }
      />
    </View>
    <Text
      style={
        [
          {
            "color": "#001D51",
            "fontFamily": "Outfit-Bold",
            "fontSize": 42,
          },
          {
            "marginBottom": 24,
          },
        ]
      }
    >
      login.signUp
    </Text>
    <View
      style={
        [
          {
            "marginBottom": 26.285714285714285,
            "width": "100%",
          },
          undefined,
        ]
      }
    >
      <Text
        style={
          [
            {
              "color": "#0A0F0F",
              "fontFamily": "Outfit-Regular",
              "fontSize": 24,
            },
            [
              {
                "marginBottom": 13.142857142857142,
              },
              undefined,
            ],
          ]
        }
      >
        auth.name
      </Text>
      <View
        style={
          {
            "alignItems": "center",
            "flexDirection": "row",
          }
        }
      >
        <TextInput
          autoCapitalize="none"
          autoCorrect={false}
          placeholder="auth.name"
          placeholderTextColor="rgba(10, 15, 15, 0.5)"
          selectionColor="#d92029"
          style={
            [
              {
                "borderColor": "#0A0F0F1A",
                "borderRadius": 12,
                "borderWidth": 1.5,
                "color": "#0A0F0F",
                "flex": 1,
                "fontFamily": "Outfit-Medium",
                "fontSize": 24,
                "padding": 18,
              },
              undefined,
            ]
          }
        />
      </View>
    </View>
    <View
      style={
        [
          {
            "marginBottom": 26.285714285714285,
            "width": "100%",
          },
          undefined,
        ]
      }
    >
      <Text
        style={
          [
            {
              "color": "#0A0F0F",
              "fontFamily": "Outfit-Regular",
              "fontSize": 24,
            },
            [
              {
                "marginBottom": 13.142857142857142,
              },
              undefined,
            ],
          ]
        }
      >
        login.email
      </Text>
      <View
        style={
          {
            "alignItems": "center",
            "flexDirection": "row",
          }
        }
      >
        <TextInput
          autoCapitalize="none"
          autoCorrect={false}
          keyboardType="email-address"
          onChangeText={[MockFunction]}
          placeholder="login.email"
          placeholderTextColor="rgba(10, 15, 15, 0.5)"
          selectionColor="#d92029"
          style={
            [
              {
                "borderColor": "#0A0F0F1A",
                "borderRadius": 12,
                "borderWidth": 1.5,
                "color": "#0A0F0F",
                "flex": 1,
                "fontFamily": "Outfit-Medium",
                "fontSize": 24,
                "padding": 18,
              },
              undefined,
            ]
          }
        />
      </View>
    </View>
    <View
      style={
        [
          {
            "marginBottom": 26.285714285714285,
            "width": "100%",
          },
          undefined,
        ]
      }
    >
      <Text
        style={
          [
            {
              "color": "#0A0F0F",
              "fontFamily": "Outfit-Regular",
              "fontSize": 24,
            },
            [
              {
                "marginBottom": 13.142857142857142,
              },
              undefined,
            ],
          ]
        }
      >
        auth.createPassword
      </Text>
      <View
        style={
          {
            "alignItems": "center",
            "flexDirection": "row",
          }
        }
      >
        <TextInput
          autoCapitalize="none"
          autoCorrect={false}
          onChangeText={[MockFunction]}
          placeholder="auth.createPassword"
          placeholderTextColor="rgba(10, 15, 15, 0.5)"
          secureTextEntry={true}
          selectionColor="#d92029"
          style={
            [
              {
                "borderColor": "#0A0F0F1A",
                "borderRadius": 12,
                "borderWidth": 1.5,
                "color": "#0A0F0F",
                "flex": 1,
                "fontFamily": "Outfit-Medium",
                "fontSize": 24,
                "padding": 18,
              },
              undefined,
            ]
          }
        />
      </View>
    </View>
    <View
      style={
        [
          {
            "marginBottom": 26.285714285714285,
            "width": "100%",
          },
          undefined,
        ]
      }
    >
      <Text
        style={
          [
            {
              "color": "#0A0F0F",
              "fontFamily": "Outfit-Regular",
              "fontSize": 24,
            },
            [
              {
                "marginBottom": 13.142857142857142,
              },
              undefined,
            ],
          ]
        }
      >
        auth.confirmPassword
      </Text>
      <View
        style={
          {
            "alignItems": "center",
            "flexDirection": "row",
          }
        }
      >
        <TextInput
          autoCapitalize="none"
          autoCorrect={false}
          onChangeText={[MockFunction]}
          placeholder="auth.confirmPassword"
          placeholderTextColor="rgba(10, 15, 15, 0.5)"
          secureTextEntry={true}
          selectionColor="#d92029"
          style={
            [
              {
                "borderColor": "#0A0F0F1A",
                "borderRadius": 12,
                "borderWidth": 1.5,
                "color": "#0A0F0F",
                "flex": 1,
                "fontFamily": "Outfit-Medium",
                "fontSize": 24,
                "padding": 18,
              },
              undefined,
            ]
          }
        />
      </View>
    </View>
    <View
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": true,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={false}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "alignItems": "center",
          "backgroundColor": "#d92029",
          "borderRadius": 12,
          "flexDirection": "row",
          "justifyContent": "center",
          "marginTop": 39.42857142857143,
          "marginVertical": 12,
          "opacity": 0.6,
          "paddingHorizontal": 42,
          "paddingVertical": 24,
          "width": "100%",
        }
      }
      testID="button-root"
    >
      <Text
        style={
          [
            {
              "color": "#0A0F0F",
              "fontFamily": "Outfit-Regular",
              "fontSize": 24,
            },
            [
              {
                "color": "#FFFFFF",
                "fontFamily": "Outfit-Bold",
                "marginHorizontal": 12,
              },
              undefined,
            ],
          ]
        }
      >
        login.signUp
      </Text>
    </View>
    <Text
      onPress={[Function]}
      style={
        [
          {
            "color": "#0A0F0F",
            "fontFamily": "Outfit-Regular",
            "fontSize": 24,
          },
          {
            "color": "#001D51",
            "fontSize": 24,
            "marginTop": 13.142857142857142,
            "textAlign": "center",
          },
        ]
      }
    >
      login.alreadyAccount
       
      <Text
        style={
          [
            {
              "color": "#0A0F0F",
              "fontFamily": "Outfit-Regular",
              "fontSize": 24,
            },
            {
              "fontFamily": "Outfit-Medium",
            },
          ]
        }
      >
        login.signIn
      </Text>
    </Text>
  </View>
</RNCSafeAreaView>
`;
